crypto/asn1/ameth_lib.c

```diff
@@ -417,3 +417,19 @@ void EVP_PKEY_asn1_set_set_pub_key(EVP_PKEY_ASN1_METHOD *ameth,
 {
     ameth->set_pub_key = set_pub_key;
 }
+
+void EVP_PKEY_asn1_set_get_priv_key(EVP_PKEY_ASN1_METHOD *ameth,
```

Comment:
Why is "asn1" in the name?

Reply:
This is a function to help set up a new custom EVP_PKEY_ASN1_METHOD. Hence the ASN1 in the name (as is consistent with all the other functions above this). The name of the method itself is a misnomer. Originally this was about encoding and decoing to/from ASN.1. This has changed over time to encompass much more. Now its about constructing, checking, encoding, decoding and setting up keys and parameters ready for later use (which may include some ASN.1 work). It probably should have a new name, but that's how it is in our API so we're stuck with it for now.